cmd/jag/commands/envelope.go

```diff
@@ -41,11 +53,10 @@ func isURL(path string) bool {
 
 func downloadGzipped(ctx context.Context, url string, path string) error {
 	bundle, err := download(ctx, url)
-	defer bundle.Close()
 	if err != nil {
 		return err
 	}
-
+	defer bundle.Close()
```

Comment:
Uh.. this seems unrelated, right?

Also (I'm not an expert at golang so please correct me) can deferring after the `err != nil check` lead to bundle not being closed because in case of an error the program will return before the defer is set?

Reply:
The thinking is that if `err == nil`, then `bundle` will be nil or at least not open. If `bundle` is nil, then calling `bundle.Close` (even deferred) will cause a crash.